packages/vite/src/node/server/index.ts

```diff
@@ -745,15 +758,16 @@ async function restartServer(server: ViteDevServer) {
 
 async function updateCjsSsrExternals(server: ViteDevServer) {
   if (!server._ssrExternals) {
-    // We use the non-ssr optimized deps to find known imports
     let knownImports: string[] = []
-    const depsOptimizer = getDepsOptimizer(server.config)
+
+    // Important! We use the non-ssr optimized deps to find known imports
+    const depsOptimizer = getDepsOptimizer(server.config, { ssr: false })
```

Comment:
Perhaps we can explain why we use non-ssr optimized deps here. I'm actually not sure too 🤔

Reply:
We use them because in v2, we only had these deps. Actually we only need the list of deps for known imports. During ssr true we don't have that list, we only optimize the explicitly defined deps right now. And I don't think we should touch how legacy cjs externalization heuristics work